main.tf

```diff
@@ -228,6 +228,25 @@ resource "aws_route_table" "private" {
 
   vpc_id = local.vpc_id
 
+  dynamic "route" {
```

Comment:
Unfortunately, adding the `route` element here will conflict with the rules that are created if a NAT gateway is also created. See the warning at the top of this page: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route_table
> NOTE on Route Tables and Routes:
>
> Terraform currently provides both a standalone [Route resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route) and a Route Table resource with routes defined in-line. At this time you cannot use a Route Table with in-line routes in conjunction with any Route resources. Doing so will cause a conflict of rule settings and will overwrite rules.

In order to support both NAT gateway routes on the private route tables, and the ones provided via the new parameter, you would need to use an `aws_route` resource and `for_each` over the private route tables while also applying the change to every route table created (one per nat gateway).

For example, it would need something like this (illustrative example only):

```
resource "aws_route" "private_route" {
  # setproduct of two sets gives a pair, so each.value[0] is the route table ID and
  # each.value[1] is a mapping of route table values.
  for_each = setproduct(aws_route_table.private[*].id, var.private_route_table_routes_extra)
  
  route_table_id = each.value[0]  # refers to aws_route_table.private[*].id

  destination_cidr_block = each.value[1].destination_cidr_block
  ipv6_cidr_block = lookup(each.value[1], "destination_ipv6_cidr_block", null)
  ... # and so on
}
```